src/Raven.Server/RavenServer.cs

```diff
@@ -1007,13 +1008,34 @@ public TcpListenerStatus StartTcpListener()
             return status;
         }
 
+        private bool PortInUse(int port)
+        {
+            var ipGlobalProperties = IPGlobalProperties.GetIPGlobalProperties();
+            return ipGlobalProperties.GetActiveTcpListeners().FirstOrDefault(endPoint => endPoint.Port == port) != null;
+        }
+
         private void StartListeners(string host, int port, TcpListenerStatus status)
         {
             try
             {
                 _tcpAuditLog = LoggingSource.AuditLog.IsInfoEnabled ? LoggingSource.AuditLog.GetLogger("TcpConnections", "Audit") : null;
                 bool successfullyBoundToAtLeastOne = false;
                 var errors = new List<Exception>();
+
+                if (PortInUse(port))
```

Comment:
Why do the check instead of letting the OS do that? Better to handle the error in this case.

Reply:
The OS will throw an error only if will try to connect the TCP